<launch>

  <!-- gazebo args -->
  <arg name="gazebo_gui"    default="false" />
  <arg name="rviz"          default="true" />
  <arg name="world"         default="$(find intel_aero_gazebo)/worlds/terrain.world" />
  <arg name="verbose"       default="false" />
  <arg name="paused"        default="false" />
  <arg name="gazebo_debug"  default="false" />

  <!-- robot args -->
  <arg name="static_map"         default="true"/>
  <arg name="static_map_file"    if="$(arg static_map)"     default="$(find intel_aero_gazebo)/maps/terrain_no_ground.bt"/>
  <arg name="static_map_topic"   if="$(arg static_map)"     value="/projected_map"/>
  <arg name="static_map_topic"   unless="$(arg static_map)" value="projected_map"/>
  <arg name="world_frame"        default="world"/>
  <arg name="robot_radius"       default="1.0"/>
  <arg name="inflation_radius"   default="1.0"/>
  <arg name="octomap_resolution" default="0.5"/>

  <!-- launch gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="gui"         value="$(arg gazebo_gui)" />
    <arg name="world_name"  value="$(arg world)" />
    <arg name="verbose"     value="$(arg verbose)" />
    <arg name="paused"      value="$(arg paused)" />
    <arg name="debug"       value="$(arg gazebo_debug)" />
  </include>

  <!-- if enabled, load saved octomap -->
  <node if="$(arg static_map)"
        pkg="octomap_server"
        type="octomap_server_node"
        name="static_map_node"
        args="$(arg static_map_file)">
    <remap from="projected_map" to="$(arg static_map_topic)"/>
    <param name="frame_id" value="$(arg world_frame)"/>
  </node>

  <!-- rviz -->
  <node if="$(arg rviz)"
    name="rviz"
    pkg="rviz"
    type="rviz"
    args="-d $(find intel_aero_gazebo)/rviz/swarm7.rviz"/>

  <!-- network topology visualization
  <node pkg="network_status"
        type="network_visualization_node"
        name="network_visualization_node"
        output="screen">
    <param name="number_of_agents" value="4"/>
    <param name="task_agent_id"    value="4"/>
    <param name="world_frame"      value="$(arg world_frame)"/>
  </node>
  -->

</launch>
